### ISSUE TITLE
[bug] HTML characters should be allowed inside of `no-useless-fragment`

### ISSUE THREAD

**Issue by imjordanxd** [Rel1cx/eslint-react#850]
### Describe the bug

The following code produces an error but is a false positive:
```tsx
function Foo() {
	return <>&nbsp;</>;
}
```

### Expected behavior

An error should not be reported in this case.

### Platform and versions

```Text
Node 20.18.0
@eslint-react 1.15.0
```

**Comment by SukkaW** [comment]
Any more reproduction?

With React 18.3+ and TypeScript 5.4+, it is possible to return a `string` in a Component now. Your current reproduction can be simplified to:

```tsx
function Foo() {
	return '&nbsp;';
}
```

**Comment by imjordanxd** [comment]
@SukkaW no, unfortunately that won't work. `&nbsp;` is a reserved character in HTML. React handles strings for us. It's the same reason we can't return `'<div>Hello, world</div>'`. We wouldn't expect that to work! [Here's a codesandbox](https://codesandbox.io/p/sandbox/yt8ccp?file=/src/App.tsx:4,1).

**Comment by SukkaW** [comment]
> ```tsx
> function Foo() {
>   return '&nbsp;';
> }
> ```

Then it would be:

```tsx
function Foo() {
  return ' ';
}
```

**Comment by rEl1cx** [comment]
ref: https://github.com/biomejs/biome/issues/4751#issuecomment-2564818839
